skyshop/src/main/java/org/skypro/skyshop/controller/ShopControllerAdvice.java

```diff
@@ -0,0 +1,17 @@
+package org.skypro.skyshop.controller;
+
+import org.skypro.skyshop.model.exceptions.ShopError;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+@RestControllerAdvice
+public class ShopControllerAdvice {
+    @ExceptionHandler(NoSuchFieldException.class)
+    public ResponseEntity<ShopError> handleException(NoSuchFieldException e) {
```

Comment:
Достаточно чего-то одного: либо указать класс исключения, либо параметр метода

```suggestion
    @ExceptionHandler
    public ResponseEntity<ShopError> handleException(NoSuchFieldException e) {
```